packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -196,42 +196,57 @@ class FlutterPlugin implements Plugin<Project> {
                         compile pluginProject
                     }
                 }
-		pluginProject.afterEvaluate {
-                    pluginProject.android.buildTypes {
+                pluginProject.afterEvaluate {
+                    pluginProject.android.buildTypes { buildTypes ->
                         profile {
                             initWith debug
                         }
+
+                        buildTypes.each {
+                            this.addFlutterJarCompileOnlyDependency(pluginProject, it)
+                        }
                     }
-		}
-                pluginProject.afterEvaluate this.&addFlutterJarCompileOnlyDependency
+                }
             } else {
                 project.logger.error("Plugin project :$name not found. Please update settings.gradle.")
             }
         }
     }
 
-    private void addFlutterJarCompileOnlyDependency(Project project) {
+    /**
+     * Adds suitable flutter.jar api dependencies to the specified buildType.
+     *
+     * Note: The BuildType DSL type is not public, and is therefore omitted from the signature.
+     */
+    private void addFlutterJarCompileOnlyDependency(Project project, buildType) {
```

Comment:
I'm unsure if this method was originally designed to handle non-Android plugin projects. Is that a valid usecase?

These changes are currently dependent on the plugin being an Android project.